dataset/metadata.go

```diff
@@ -0,0 +1,257 @@
+package dataset
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Metadata is the structure of dataset metadata
+type Metadata struct {
+	ID                  string     `json:"id"`
+	Name                string     `json:"name"`
+	Description         string     `json:"description"`
+	CreatedAt           *time.Time `json:"created_at"`
+	CreatedBy           string     `json:"created_by"`
+	DataClassifications []string   `json:"data_classifications"`
+	DataFormat          string     `json:"data_format"`
+	Derivative          bool       `json:"derivative"`
+	DuaURL              *url.URL   `json:"dua_url"`
+	ModifiedAt          *time.Time `json:"modified_at"`
+	ModifiedBy          string     `json:"modified_by"`
```

Comment:
Is this for modifying the dataset or the metadata?
Maybe we need to make this a list `Modifications` so it can have multiple `modified_at` and `modified_by` entries.

Reply:
This is supposed to be the dataset (documented on that confluence page).  You might be right about keeping a list... at this point however, I'm not even sure how we'll track it.